app/src/main/java/ch/hikemate/app/ui/map/RunHikeScreen.kt

```diff
@@ -68,30 +67,80 @@ object RunHikeScreen {
 
 @Composable
 fun RunHikeScreen(
-    listOfHikeRoutesViewModel: ListOfHikeRoutesViewModel,
-    profileViewModel: ProfileViewModel,
+    hikesViewModel: HikesViewModel,
     navigationActions: NavigationActions,
 ) {
+  val selectedHike by hikesViewModel.selectedHike.collectAsState()
 
-  val context = LocalContext.current
-
-  val selectedRoute by listOfHikeRoutesViewModel.selectedHikeRoute.collectAsState()
-
-  LaunchedEffect(selectedRoute) {
-    if (selectedRoute == null) {
+  LaunchedEffect(selectedHike) {
+    if (selectedHike == null) {
       navigationActions.goBack()
     }
   }
-  val route = selectedRoute!!
 
-  // This will need to be changed when the "run" feature of this screen is implemented
-  val routeZoomLevel = MapUtils.calculateBestZoomLevel(route.bounds).toDouble()
+  // Can't display the details without a selected hike
+  if (selectedHike == null) {
+    return
+  }
+
+  val hike = selectedHike!!
+
+  WithDetailedHike(
+      hike = hike,
+      hikesViewModel = hikesViewModel,
+      withDetailedHike = { RunHikeContent(it, navigationActions) },
+      whenError = {
+        CenteredErrorAction(
+            errorMessageId = R.string.loading_hike_error,
+            actionIcon = Icons.AutoMirrored.Filled.ArrowBack,
+            actionContentDescriptionStringId = R.string.go_back,
+            onAction = { navigationActions.goBack() })
+      })
+}
+
+@Composable
+private fun RunHikeContent(hike: DetailedHike, navigationActions: NavigationActions) {
+  // Avoids the app crashing when spamming the back button
+  var wantToNavigateBack by remember { mutableStateOf(false) }
+  LaunchedEffect(wantToNavigateBack) { if (wantToNavigateBack) navigationActions.goBack() }
```

Comment:
I'm not sure to understand what the observed bug is, could you maybe clarify?

`wantToNavigateBack`'s purpose is precisely to avoid spamming being a problem. With this, `navigationActions.goBack()` is called only once.

Reply:
Sorry my explanation wasn't clear enough. Basically you can reproduce it doing the following:
Going very quickly between the `HikeDetailsScreen` and the `RunHikeScreen` by clicking on the _Run this hike_ button and _Stop the run_ **over and over** you'll see that if you do it quickly enough that the _Run this hike_ button won't work anymore after two-three times.
To solve that you need to force the value of the `wantToNavigateBack` and `wantToRunHike` to false after doing the action.
This is not the same problem as you stated but I should've explained it better.

Reply:
Ohhh I see. I was able to reproduce, I don't know why but I mistook this bug with another one, the one of the back button being spammed. Anyway, this should now be fixed !